lib/rspec/support/object_inspector.rb

```diff
@@ -13,6 +13,8 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
+          "Double (#{object.instance_variable_get(:@name)})"
```

Comment:
@myronmarston As per your comment I understood following points:
 
1) We need to define a method called `ObjectInspector.register(double)` in `ObjectInspector` class.
2) The above method will be called from `rspec-mocks` to register doubles.
3)  Return value of above method would be `Double (Foo)`.

Please correct me if I understood anything wrong.